fibers/operations.scm

```diff
@@ -121,6 +127,18 @@ succeeds, will succeed with one and only one of the sub-operations
     ((base-op) base-op)
     (base-ops (make-choice-operation (list->vector base-ops)))))
 
+(define (cancel-other-operations op index)
```

Comment:
If you do the thing I mentioned in `timer-operation`, this is going to need a state argument.